### ISSUE TITLE
api: duplicate code for VAULT_AGENT_ADDR and VAULT_SKIP_VERIFY

### ISSUE THREAD

**Issue by candlerb** [hashicorp/vault#11055]
**Describe the bug**
This is a very minor niggle, but in api/client.go there are cases where the same environment variable is read twice:

```
const EnvVaultAgentAddr = "VAULT_AGENT_ADDR"
const EnvVaultSkipVerify = "VAULT_SKIP_VERIFY"
...
// Deprecated values
const EnvVaultAgentAddress = "VAULT_AGENT_ADDR"
const EnvVaultInsecure = "VAULT_SKIP_VERIFY"
...
        if v := os.Getenv(EnvVaultAgentAddr); v != "" {
                envAgentAddress = v
        } else if v := os.Getenv(EnvVaultAgentAddress); v != "" {
                envAgentAddress = v
        }
...
        if v := os.Getenv(EnvVaultSkipVerify); v != "" {
                var err error
                envInsecure, err = strconv.ParseBool(v)
                if err != nil {
                        return fmt.Errorf("could not parse VAULT_SKIP_VERIFY")
                }
        } else if v := os.Getenv(EnvVaultInsecure); v != "" {
                var err error
                envInsecure, err = strconv.ParseBool(v)
                if err != nil {
                        return fmt.Errorf("could not parse VAULT_INSECURE")
                }
        }
```

The error message suggests there was a `VAULT_INSECURE`, but I don't think that's true.  AFAICS from git history, there have never been `VAULT_INSECURE` or `VAULT_AGENT_ADDRESS` environment variables.

However, the exported `const` names `EnvVaultXXX` changed in PR #6306.  Then commit 61a17292 re-introduced them, but also added the duplicate code (which seems to be unnecessary).

**Suggestion**

Change to
```
// Deprecated values
const EnvVaultAgentAddress = EnvVaultAgentAddr
const EnvVaultInsecure = EnvVaultSkipVerify
```

and then remove the duplicate code.

**Comment by aphorise** [comment]
Although I was not able to correlate the exact commit in which this was found - it does seem to be correct in the [current code: vault/api/client.go](https://github.com/hashicorp/vault/blob/main/api/client.go#L387-L393)

Hey @candlerb good catch - do you agree that it's been fixed and ok to close?

**Comment by candlerb** [comment]
Looks like this was fixed in d5305500:

> * Removed redundant checks for same env var in ReadEnvironment

This is good to close, thank you.
